sql/query/expr.go

```diff
@@ -387,7 +399,9 @@ func evalComparisonExpr(expr *parser.ComparisonExpr, env Env) (sqlwire.Datum, er
 }
 
 // Prepare the arguments for a binary operation. The returned arguments will
-// have the same type.
+// have the same type. The typ parameter specifies the allowed types for the
+// operation. For example, bit-operations should specify intOp or uintOp to
+// indicate that they do not operate on floating point arguments.
```

Comment:
maybe add something like:
// Float operations may still reduce to intOp or uintOp if the operands support it.